js/src/common/components/Modal.tsx

```diff
@@ -78,70 +85,59 @@ export default class Modal extends Component {
 
   /**
    * Get the class name to apply to the modal.
-   *
-   * @return {String}
-   * @abstract
    */
-  className() {}
+  abstract className(): string;
```

Comment:
Not sure how I feel about making these abstract... if an extension doesn't implement any of these methods it'll break - I envision this happening mostly with `className`. If they have a subclass that is also treated as an abstract Modal class, then that would also break.

What I'm saying is, this is a breaking change 🙃